airbyte-integrations/connectors/source-zuora/source_zuora/zuora_client.py

```diff
@@ -0,0 +1,270 @@
+        # case where we have 1 sec difference between start and end dates would produce 0 slices, so return 1 instead
+        d = pendulum.period(start_date, end_date).in_days()
+        return 1 if ceil(d / window_days) <= 0 else ceil(d / window_days)
+
+    # Creates the next-date-slice
+    @staticmethod
+    def _next_slice_end_date(end_date: str, window_days: int) -> str:
+        format = "YYYY-MM-DD HH:mm:ss Z"
+        return pendulum.from_format(end_date, format).add(days=window_days).to_datetime_string()
+
+    # Get/Set the timezone information from the date_field
+    @staticmethod
+    def _get_tz(start_date: datetime) -> str:
+        return " +00:00" if start_date.tzname() == "UTC" else f" {start_date.tzname()}"
+
+    # Compares strings for datatypes casting
+    @staticmethod
+    def _check_data_type(_str, _list):
+        # Returns True if the Type is in the List of Types, else - False
+        return any(_str in s for s in _list)
+
+    # Convert Zuora DataTypes to JsonSchema Types
+    def _cast_schema_types(self, _schema: List[Dict]) -> Dict:
+        casted_schema_types = {}
+
+        for field in range(len(_schema)):
+            field_type = _schema[field].get("type").lower()
+
+            # Casting Zuora ZOQL Data Types into JsonSchema types
+            if self._check_data_type(field_type, ["decimal(22,9)", "integer", "number", "int", "bigint", "smallint", "timestamp"]):
+                field_type = ["number", "null"]
+            elif self._check_data_type(field_type, ["date", "datetime", "timestamp with time zone", "picklist", "text", "varchar"]):
+                field_type = ["string", "null"]
+            elif self._check_data_type(field_type, ["zoql", "binary", "json", "xml", "blob"]):
+                field_type = ["object", "null"]
+            elif self._check_data_type(field_type, ["list", "array"]):
+                field_type = ["array", "null"]
+            elif self._check_data_type(field_type, ["boolean"]):
+                field_type = ["boolean", "null"]
+            else:
+                # if there is something else we don't cover, cast it as string
+                field_type = ["string", "null"]
+
+            casted_schema_types.update(**{_schema[field].get("field").lower(): {"type": field_type}})
+
+        return casted_schema_types
+
+    # Convert Zuora Fields data types to JSONSchema
+    def _zuora_object_to_json_schema(self, obj: str) -> Dict:
+        self.logger.info(f"Getting schema information for {obj}")
+        raw_data_types = list(self._get_data(q_type="describe", obj=obj))
+        json_schema = []
+        # Get the raw field names and their types
+        for field in range(len(raw_data_types)):
+            json_schema.append({"field": raw_data_types[field]["Column"], "type": raw_data_types[field]["Type"]})
+        # Cast the Zuora Field Types to JsonSchema Types
+        json_schema = self._cast_schema_types(json_schema)
+        return json_schema
```

Comment:
```suggestion
       json_schema = [{"field": data_type["Column"], "type": data_type["Type"]} for data_type in raw_data_types]
       return self._cast_schema_types(json_schema)
```

Reply:
+++

Reply:
or even better :
       return = [{"field": data_type["Column"], "type": covert_type(data_type["Type"])} for data_type in raw_data_types]

where 'covert_type' - is similar 'to _cast_schema_types'

Reply:
Thanks